ruby/meter-health/meter_health.rb

```diff
@@ -0,0 +1,117 @@
+  METERS = "/api/v2/organizations/%d/meters.json"
+  SITE_BASE_URL = 'http://54.90.173.204:8080'
+  SITE_OAUTH_PATH = '/oauth/token'
+
+  def initialize
+    @credentials = retrieve_credentials
+    @token = generate_token
+  end
+
+  def create_report
+    if @token
+      request_information
+    else
+      puts "Verify yor credentials, we weren't able to retrieve a token"
+    end
+  end
+
+  private
+  
+  def retrieve_credentials
+    file = File.read('./config/credentials.json')
+    JSON.parse(file)
+  end
+
+  def generate_token
+    client = OAuth2::Client.new(@credentials["app_id"], @credentials["app_secret"],
+      site: SITE_BASE_URL, token_url: SITE_OAUTH_PATH, scope: @credentials["scope"])
+    token = client.password.get_token(@credentials["user_email"],
+      @credentials["user_password"],scope: @credentials["scope"])
+    token and token.token ? token.token : nil
+  end
+
+  def request_information
+    organizations = JSON.parse(retrieve_organizations)
+    if organizations && organizations["embedded"]["organizations"]
+      write_to_csv(organizations["embedded"]["organizations"])
+    else
+      puts "There are not organizations"
+    end
+  end
+
+  def retrieve_organizations
+    response = RestClient.get("#{SITE_BASE_URL}#{ORGANIZATIONS}",
+      { params: { access_token: @token, expand: 'organizations' } })
+  end
+
+  def write_to_csv(organizations)
+    CSV.open(@credentials["file_name"], "w+") do |csv|
+      csv << ['Organization Name', 'Meter ID', 'Meter Name',
+              'Meter Type', 'Meter Status', 'Last Reading']
+      total_org = organizations.count
+      org_index = 0
+      organizations.each do |org|
+        print "Retrieving meters for organization #{org_index} of #{total_org}\n"
+        retrieve_and_write_meters_to_csv(org,csv)
+        org_index += 1
+      end
+    end
+  end
+
+  def retrieve_and_write_meters_to_csv(org,csv)
+    meters = retrieve_meters_for_org(org)
+    meters.each do |meter|
+      csv << [ org['name'],
+               meter['remote_id'],
+               meter['name'],
+               meter['kind'],
+               meter['status'],
+               meter['last_processed_on']
+             ]
+    end
+  end
+
+  def retrieve_meters_for_org(org)
+    response = request_meters(org)
+    meters = JSON.parse(response)
+    if meters && meters["embedded"]["meters"]
+      meters["embedded"]["meters"]
+    else
+      []
+    end
+  end
+
+  def request_meters(org)
+    tries = 20
+    begin
+      response = RestClient.get(("#{SITE_BASE_URL}#{METERS}" % org['remote_id']),
+      { params: { access_token: @token } })
+      return response
+    rescue
+      puts "Reached the total requests allowed in a minute...\n Waiting to retry\n"
+      tries -= 1
+      if tries > 0
+        sleep(5)
+        puts "Retrying for org #{org['name']}\n"
+        retry
+      else
+        puts "We were not able to retrieve info for org #{org['name']} there have being a lot of tries"
```

Comment:
You could replace this: `there have being a lot of tries` with something like `too many retries attempted`.